pkg/cvo/internal/operatorstatus_test.go

```diff
@@ -0,0 +1,217 @@
+package internal
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	clientgotesting "k8s.io/client-go/testing"
+
+	configv1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/client-go/config/clientset/versioned/fake"
+	"github.com/pkg/errors"
+)
+
+func Test_waitForOperatorStatusToBeDone(t *testing.T) {
+	expected := &configv1.ClusterOperator{
+		ObjectMeta: metav1.ObjectMeta{Name: "test-co"},
+		Status: configv1.ClusterOperatorStatus{
+			Versions: []configv1.OperandVersion{{
+				Name: "operator", Version: "v1",
+			}, {
+				Name: "operand-1", Version: "v1",
+			}},
+		},
+	}
+	tests := []struct {
+		name   string
+		actual *configv1.ClusterOperator
+
+		wantErr string
+	}{{
+		name: "cluster operator not found",
+		actual: &configv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{Name: "dummy"},
+		},
+		wantErr: "clusteroperator \"test-co\" not found",
+	}, {
+		name: "cluster operator reporting no versions",
+		actual: &configv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-co"},
+		},
+		wantErr: "cluster operator test-co has not completely rolled its operands to desired version: no operand-1 at v1, no operator at v1",
+	}, {
+		name: "cluster operator reporting old versions",
+		actual: &configv1.ClusterOperator{
+			ObjectMeta: metav1.ObjectMeta{Name: "test-co"},
+			Status: configv1.ClusterOperatorStatus{
+				Versions: []configv1.OperandVersion{{
+					Name: "operator", Version: "v0",
+				}, {
+					Name: "operand-1", Version: "v0",
+				}},
+			},
+		},
+		wantErr: "cluster operator test-co has not completely rolled its operands to desired version: still rolling operand-1 from v0 to v1, still rolling operator from v0 to v1",
```

Comment:
Struggling with the wording here.  operands is fairly obscure, and the "rolling" is also potentially the wrong verb

A couple of possible rewordings:

> cluster operator test-co is still updating: operand-1 from 4.0.1 to 4.0.2, operator from 4.0.1 to 4.02

or treat operator special if it exists:

> cluster operator test-co is still updating to 4.0.2: upgrading kubernetes from 1.12.4 to 1.12.5

or just skip the operator completely, assuming that it is already known to be updating:

> cluster operator test-co is still updating: upgrading kubernetes from 1.12.4 to 1.12.5

Reply:
or maybe even special case the non operator, single operand case

> cluster operator kube-apiserver is still upgrading kubernetes from 1.12.4 to 1.12.5

Reply:
@smarterclayton went with this option:
> or just skip the operator completely, assuming that it is already known to be updating:
> 
> > cluster operator kube-apiserver is still updating: upgrading kubernetes from 1.12.4 to 1.12.5

PTAL